platform.js

```diff
@@ -423,5 +423,16 @@ module.exports = {
         });
       }
     });
+  },
+  runPromise(promise, retries) {
+    return promise()
+      .catch((err)=>{
+        if(retries > 0) {
+          return module.exports.runPromise(promise, retries - 1);
```

Comment:
@fjvallarino can we add a timeout here? As for the timeout interval, I'm not sure what number we should go with, but we should make it take long enough so the display has a chance to reconnect (not necessarily on the first attempt, but after a few attempts)